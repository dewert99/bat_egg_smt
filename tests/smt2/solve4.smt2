(declare-sort S1)
(declare-sort S2 1)
(declare-const a S1)
(declare-fun b () S1)
(declare-fun f (S1) (S2 S1))
(assert (not (= (f a) (f b))))
(check-sat-assuming ((= a b)))
(get-unsat-core)
(check-sat)
(get-value (a b (f a) (f b)))