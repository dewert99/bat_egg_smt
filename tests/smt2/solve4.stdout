unsat
((= a b))
sat
(
 (define-fun a () S1 (as @v3 S1))
 (define-fun b () S1 (as @v4 S1))
 (define-fun f ((x!0 S1) (x!1 S1)) (S2 S1)
  (ite (and (= x!0 (as @v3 S1)) (= x!1 (as @v3 S1))) (as @v5 (S2 S1))
  (ite (and (= x!0 (as @v3 S1)) (= x!1 (as @v4 S1))) (as @v6 (S2 S1))
   (as @f!default (S2 S1)))))
)
((a (as @v3 S1))
 (b (as @v4 S1))
 ((f a a) (as @v5 (S2 S1)))
 ((f a b) (as @v6 (S2 S1))))
